status.go

```diff
@@ -298,6 +302,52 @@ func initStatus(services []*service) {
 		io.Copy(w, &buf)
 	})
 
+	http.HandleFunc("/log", func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Access-Control-Allow-Origin", "*")
+
+		evtStream := eventStreamRequested(r)
+		if evtStream {
+			w.Header().Set("Content-type", "text/event-stream")
+		}
+
+		path := r.FormValue("path")
+		svc := findSvc(path)
+		if svc == nil {
+			http.Error(w, "service not found", http.StatusNotFound)
+			return
+		}
+
+		streamName := r.FormValue("stream")
+
+		var stream <-chan string
+		var closeFunc func()
+
+		switch streamName {
+		case "stdout":
+			stream, closeFunc = svc.Stdout.Stream()
+		case "stderr":
+			stream, closeFunc = svc.Stderr.Stream()
+		default:
+			http.Error(w, "stream not found", http.StatusNotFound)
+			return
+		}
+		defer closeFunc()
+
+		for line := range stream {
```

Comment:
Can you tie this loop to the HTTP request context please? That way, we avoid goroutines piling up when the client has closed its connection but nothing is happening on the `stream` channel.

```go
ctx := r.Context()
for {
select {
case <-ctx.Done():
  return
case line := <-stream:
  // …
}
}
```

Reply:
Good call - done